File: CS/GridControlDetailView/CustomGrid/CustomSelectionInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraGrid.Views.Base.ViewInfo;

namespace CustomGrid
{
    public class CustomSelectionInfo : BaseSelectionInfo
    {
        public CustomSelectionInfo(CustomView view)
            : base(view)
        {
        }

        protected override int GetState()
        {
            throw new NotImplementedException();
        }
    }
}

File: VB/GridControlDetailView/CustomGrid/CustomSelectionInfo.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraGrid.Views.Base.ViewInfo

Namespace CustomGrid
	Public Class CustomSelectionInfo
		Inherits BaseSelectionInfo

		Public Sub New(ByVal view As CustomView)
			MyBase.New(view)
		End Sub

		Protected Overrides Function GetState() As Integer
			Throw New NotImplementedException()
		End Function
	End Class
End Namespace
